### PR TITLE
Allow multiple checkCanvasRectColor checks with one readPixels call

### PR DESCRIPTION
**Pull request by hendrikwagenaar**
Quick optimization to make the test run faster.  readPixels can be slow, so call readPixels once on a union of the rects, and check each subrect for matching color.

### CODE REVIEW

File: sdk/tests/conformance/resources/webgl-test-utils.js [hunk 1]
```diff
@@ -1071,6 +1071,79 @@ var isWebGLContext = function(ctx) {
...
+ */
+var makeCheckRect = function(x, y, width, height, color, msg, errorRange) {
+  var rect = {
+    'x': x, 'y': y,
+    'width': width, 'height': height,
+    'color': color, 'msg': msg,
+    'errorRange': errorRange,
+
+    'checkRect': function (buf, l, b, w) {
+      for (var px = (x - l) ; px < (width - l) ; ++px) {
```
Comment (toji) [inline sdk/tests/conformance/resources/webgl-test-utils.js]: This feels wrong. It's appears to be treating "width" as "right". A simple example is that if I test a rect with x = 10, width = 2 the loop would exit without testing any pixels.

Seems like the more accurate loop would be:
`for (var px = (x - l) ; px < (x + width - l) ; ++px) {`

Same goes for height below.

**Comment by toji** [comment]
Looks good as long as the loop logic is fixed.

**Comment by hendrikwagenaar** [comment]
Good catch, fixed

**Comment by toji** [comment]
LGTM

**Pull request merged**